src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedDelegateSymbol.cs

```diff
@@ -28,14 +25,14 @@ internal sealed class SynthesizedDelegateSymbol : SynthesizedContainer
             string name,
             TypeSymbol objectType,
             TypeSymbol intPtrType,
-            TypeSymbol voidReturnTypeOpt,
+            TypeSymbol? voidReturnTypeOpt,
             int parameterCount,
-            BitVector byRefParameters)
-            : base(name, parameterCount, returnsVoid: (object)voidReturnTypeOpt != null)
+            RefKindVector refKinds)
+            : base(name, parameterCount, returnsVoid: voidReturnTypeOpt is not null)
```

Comment:
nit: Consider repeat the assertion (or equivalent) here: ` Debug.Assert(refKinds.IsNull || parameterCount == refKinds.Capacity - (returnsVoid ? 0 : 1));`